addons/lunch/static/src/js/lunch.js

```diff
@@ -0,0 +1,67 @@
+(function() {
+    "use strict";
+
+    var QWeb = openerp.web.qweb,
+        lunch = openerp.lunch = {};
+
+    lunch.previousorder = openerp.web.form.AbstractField.extend(openerp.web.form.ReinitializeWidgetMixin,{
+        template : 'lunch.PreviousOrder',
+        events: {
+        'click span.add_button': 'set_order_line',
+        },
+        init: function(){
+            this.lunch_data = {};
+            this.fields_to_read = ['product_id', 'supplier', 'note', 'price', 'category_id','currency_id'];
+            return this._super.apply(this, arguments);
+        },
+        set_value: function(value_) {
+            value_ = value_ || [];
+            this._super((value_.length >= 1 && value_[0] instanceof Array)?value_[0][2]:value_);
```

Comment:
Here is the "expanded" version ;)
`if(value_.length >= 1 && value_[0] instanceof Array) {
    value_ = value_[0][2];
}
this._super(value_);`